shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -0,0 +1,94 @@
+use opensearch::{
+    auth::Credentials,
+    cert::CertificateValidation,
+    http::Url,
+    http::{
+        response::Response,
+        transport::{SingleNodeConnectionPool, TransportBuilder},
+        StatusCode,
+    },
+    indices::IndicesExistsParts,
+    params::Refresh,
+    BulkOperation, BulkParts, OpenSearch, SearchParts,
+};
+use serde_json::{json, Value};
+use test_helpers::docker_compose::docker_compose;
+
+pub async fn index_documents(client: &OpenSearch) -> Response {
+    let index = "posts";
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&[index]))
+        .send()
+        .await
+        .unwrap();
+
+    if exists_response.status_code() == StatusCode::NOT_FOUND {
```

Comment:
branches like this in test code are problematic.
Is it reasonable to assert that the status_code is NOT_FOUND instead?